packages/icons/src/components/icon/icon.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+import * as React from "react";
+
+import { IconName, IconPartial, IIconProps } from "../../common";
+import * as Errors from "../../common/errors";
+
+let allIcons: { [key: string]: IconPartial | undefined } | null = null;
+
+if (!(global as any).BLUEPRINT_ICONS_TREE_SHAKING) {
```

Comment:
🤔 if a user never imports this component and only uses `<NamedIcon />` exports, then will this file (and `allIcons`) be tree-shook automatically? as in, do we actually need this flag or can we rely on the build tool to detect that this module is unused?

Reply:
But `Icon` is still used under the hood in other components, so it's unlikely that it's *never* imported.